src/components/voteDialog/integration.test.js

```diff
@@ -0,0 +1,134 @@
+import { step } from 'mocha-steps';
+import { expect } from 'chai';
+import { mount } from 'enzyme';
+import sinon from 'sinon';
+import { prepareStore, renderWithRouter } from '../../utils/applicationInit';
+import actionTypes from '../../constants/actions';
+import accountReducer from '../../store/reducers/account';
+import votingReducer from '../../store/reducers/voting';
+import peersReducer from '../../store/reducers/peers';
+import { accountLoggedIn } from '../../actions/account';
+import { delegatesAdded } from '../../actions/voting';
+import * as delegateApi from '../../utils/api/delegate';
+import VoteDialog from './index';
+
+const delegates = [
+  { username: 'username1', publicKey: '123HG3452245L' },
+  { username: 'username2', publicKey: '123HG3522345L' },
+];
+const unvotedDelegate = [
+  { username: 'username3', publicKey: '123HG3522445L' },
+  { username: 'username4', publicKey: '123HG3522545L' },
+];
+
+const keyCodes = {
+  arrowDown: 40,
+  arrowUp: 38,
+  enter: 13,
+  escape: 27,
+};
+
+const store = prepareStore({
+  account: accountReducer,
+  voting: votingReducer,
+  peers: peersReducer,
+});
+
+describe.only('VoteDialog integration test', () => {
+  const realAccount = {
+    address: '16313739661670634666L',
+    balance: '346215336704',
+    delegate: {},
+    multisignatures: [],
+    passphrase: 'wagon stock borrow episode laundry kitten salute link globe zero feed marble',
+    publicKey: 'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f',
+    u_multisignatures: [],
+    unconfirmedBalance: '0',
+  };
+
+  const peers = {
+    defaultPeers: [
+      'node01.lisk.io',
+      'node02.lisk.io',
+    ],
+    defaultSSLPeers: [
+      'node01.lisk.io',
+      'node02.lisk.io',
+    ],
+    defaultTestnetPeers: [
+      'testnet.lisk.io',
+    ],
+    options: {
+      name: 'Testnet',
+      testnet: true,
+      ssl: true,
+      port: 443,
+      code: 1,
+    },
+    ssl: true,
+    randomPeer: true,
+    testnet: true,
+    bannedPeers: [],
+    currentPeer: 'testnet.lisk.io',
+    port: 443,
+    nethash: {
+      'Content-Type': 'application/json',
+      nethash: 'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
+      broadhash: 'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
+      os: 'lisk-js-api',
+      version: '1.0.0',
+      minVersion: '>=0.5.0',
+      port: 443,
+    },
+  };
+
+  store.dispatch(accountLoggedIn(realAccount));
+  store.dispatch({
+    data: peers,
+    type: actionTypes.activePeerSet,
+  });
+  const wrapper = mount(renderWithRouter(VoteDialog, store));
```

Comment:
No, it shouldn't. because there is one component which will be test in each step and we don't need to mount it before each steps. if we do that , we will lose the last state.

Reply:
It sounds like you confuse `Given` step with `beforeEach` block. What you are saying is true for `beforeEach` block, but not for `Given` step. Of course, you need to declare `let wrapper` outside of the step so that you can access it in other steps.